client/src/components/PlanShowPage.js

```diff
@@ -14,18 +17,83 @@ const PlanShowPage = (props) => {
     }
   }
 
+  const fetchUser = async () => {
+    const currentUser = await getCurrentUser()
+    setUser(currentUser)
+  }
```

Comment:
you shouldnt have to get the current user, it is already being retrieved in App.js. Instead, review how to pass down the user to the PlanShowPage component. Otherwise you are making a second costly fetch, database transaction and re-render